c7n/resources/ec2.py

```diff
@@ -1176,10 +1176,23 @@ class Stop(BaseAction, StateTransitionFilter):
               - instance-state-name: running
             actions:
               - stop
+
+          - name: ec2-hibernate-instances
+            resources: ec2
+            query:
+              - instance-state-name: running
+            actions:
+              - type: stop
+                hibernate: true
     """
     valid_origin_states = ('running',)
 
-    schema = type_schema('stop', **{'terminate-ephemeral': {'type': 'boolean'}})
+    schema = type_schema(
+        'stop',
+        **{'terminate-ephemeral': {'type': 'boolean'},
+           'hibernate': {'type': 'boolean'}})
+
+    has_hibernate = jmespath.compile('[].HibernationOptions.Configured')
```

Comment:
switched from value filter composition to just using a jmespath expression to bulk evaluate.